### ISSUE TITLE
Improvements to stdlib documentation

### ISSUE THREAD

**Issue by bobbinth** [0xMiden/miden-vm#296]
While working on #294, I thought of a few potential improvements to the automatic documentation generation we have for stdlib. To list a few of them:

* We should probably introduce a new type of comment for documentation comments to make parsing simpler. Maybe something like `#!`.
* One thing that would be good to have is a way to specify some structured info about procedures. For example, whether a procedure is `unstable`, how many cycles a procedure takes to execute, if a procedure belongs to a specific category (e.g. "arithmetic operations" for u64 module). One way to do it is via some attributes within comments - e.g., something like `@cycles`. Another way could be general attributes for procedures (similar to how Rust does it).
* Support Markdown within comments (similar to Rust) so that we could have nice-looking procedure descriptions. This would also mean that we'd need to move away from outputting procedure docs as a single big table.

**Comment by tnachen** [comment]
This makes sense, I think for '#2' a general attribute notation makes sense and the parser can lay out columns based on attributes detected. The downside I suppose is that if we want to use these attributes to be more useful than just for docs display, then returning more defined attributes in the class can be useful for other purposes.

**Comment by bobbinth** [comment]
I tend to agree - there is probably a set of attributes which can be useful beyond documentation purposes (e.g., `unstable` attribute could be use during compilation to print out a warning). Though, I'm a bit hesitant to have purely informational attributes (e.g., number of cycles) handles this way too. And having two different types of attributes (one general, and one for documentation purposes) sees like a lot of complexity.

Maybe the way to figure this out is to come up with an initial list of attributes and see if there is a good way to combine them. So far, I can think of the following:

* `unstable` - to mark procedures which are already in `stdlib` but still need to be tested/optimized.
* `serialize(inline)` - to mark procedures which should be inlined during serialization. This could be useful for small procedures (e.g., when a procedure itself can be serialized in fewer than 32 bytes). May also be useful for procedures with constant parameters as described in #297.
* `num_cycles` - number of cycles a procedure requires to execute. One thing to think through is how to handle procedures where the number of cycles depends on user input.
* `doc_section` - name of a documentation section to which the procedure should be assigned (e.g., "arithmetic operations" vs. "bitwise operations" in `u64` module).

**Comment by tnachen** [comment]
Ah got it, seems like the serialize and const is actually useful for doc and also the main parser. It seems like it might be easier to treat documentation parser and the actual masm parser separate as the main metadata that's being extracted is a bit different (description and comments is not parsed, etc).

If purely doc focus feels like we can special case variables that are important for doc generation and the rest can be treated as more columns in the output table.

@bobbinth wdyt?
